### PR TITLE
Do not send an empty supported groups extension

### LINKED ISSUE #13583: If no supported_groups are available then we send an empty list
It is possible to not have any supported_groups available, and in that case a libssl client will send an empty list which is a protocol error (the supported_groups extension if sent must always have at least one element in it).

For example, if you configure OpenSSL with "no-ec2m" and then attempt to send a curve list containing only an ec2m curve: 
````
$ openssl s_client -no_tls1_3 -groups sect163k1
````

Then you get this error on the server:
````
$ openssl s_server
Using default temp DH parameters
ACCEPT
ERROR
140253113922880:error:1424206E:SSL routines:tls_parse_ctos_supported_groups:bad extension:ssl/statem/extensions_srvr.c:964:
shutting down SSL
CONNECTION CLOSED
````

Affects both master and 1.1.1

### PR DESCRIPTION
**Pull request by t8m**
This allows handshake to proceed if the maximum TLS version enabled is <1.3

Fixes #13583

##### Checklist

- [x] tests are added or updated

### CODE REVIEW

**Comment by t-j-h** [comment]
This needs a CHANGES entry - it is altering a behaviour.

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
Cleanly rebased so the CHANGES from the latest release is updated.  I've added a fixup containing the CHANGES entry as requested by Tim.

@paulidale @mattcaswell Are you still OK with the patch?

 I do not think the 24 hour period needs to be restarted.

**Review by mattcaswell — APPROVED** [review]
Still good

File: CHANGES [hunk 1]
```diff
@@ -9,20 +9,24 @@
 
  Changes between 1.1.1o and 1.1.1p [xx XXX xxxx]
 
-  *)
+  *) When OpenSSL TLS client is connecting without any supported elliptic
+     curves and TLS-1.3 protocol disabled the connection will no longer fail
```
Comment (paulidale) [inline CHANGES:13]: protocol *is* disabled

**Review by paulidale — APPROVED** [review]
One wording nit, but still good.

File: CHANGES (hunk 1 above)
Comment (t8m) [reply CHANGES]: Fixed @mattcaswell assuming this is still OK.

**Review by mattcaswell — APPROVED** [review]
Still good

**Comment by t8m** [comment]
Merged to 1.1.1 branch. Thank you for the reviews.

**Pull request closed without merging**